bigtable-hbase-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/read/TestScanAdapter.java

```diff
@@ -51,17 +54,39 @@ public ReadRowsRequest applyPreSendHook(ReadRowsRequest readRowsRequest) {
     }
   };
 
+  private static RowRange toRange(byte[] start, byte[] stop) {
+    return RowRange.newBuilder().setStartKeyClosed(ByteStringer.wrap(start))
+        .setEndKeyOpen(ByteStringer.wrap(stop)).build();
+  }
+
+  private static RowSet toRowSet(RowRange range) {
+    return RowSet.newBuilder().addRowRanges(range).build();
+  }
+
+  private static byte[] calculatePrefixEnd(byte[] prefix) {
+    byte[] prefixEnd = new byte[prefix.length];
+    System.arraycopy(prefix, 0, prefixEnd, 0, prefixEnd.length);
+    prefixEnd[prefixEnd.length - 1]++;
+    return prefixEnd;
+  }
+
   @Test
   public void testStartAndEndKeysAreSet() {
     byte[] startKey = Bytes.toBytes("startKey");
     byte[] stopKey = Bytes.toBytes("stopKey");
+    Scan scan = new Scan(startKey, stopKey);
+    ReadRowsRequest.Builder request = scanAdapter.adapt(scan, throwingReadHooks);
+    Assert.assertEquals(toRowSet(toRange(startKey, stopKey)), request.getRows());
+  }
+
+  @Test
+  public void testPrefix() {
+    byte[] prefix = Bytes.toBytes("prefix");
+    byte[] prefixEnd = calculatePrefixEnd(prefix);
     Scan scan = new Scan();
-    scan.setStartRow(startKey);
-    scan.setStopRow(stopKey);
+    scan.setRowPrefixFilter(prefix);
     ReadRowsRequest.Builder request = scanAdapter.adapt(scan, throwingReadHooks);
-    RowRange range = request.getRows().getRowRanges(0);
-    Assert.assertArrayEquals(startKey, range.getStartKeyClosed().toByteArray());
-    Assert.assertArrayEquals(stopKey, range.getEndKeyOpen().toByteArray());
+    Assert.assertEquals(toRowSet(toRange(prefix, prefixEnd)), request.getRows());
```

Comment:
Would be good to test the boundary cases for binary keys: trailing 0xff and negatives